tests/unit_tests/test_runner.py

```diff
@@ -282,3 +285,223 @@ def test_evaluate(
             )
 
     result_file.unlink()
+
+
+def test_metrics_logging(tmp_path, mgf_small, tiny_config):
+    config = Config(tiny_config)
+    config.log_metrics = True
+    config.log_every_n_steps = 1
+    config.tb_summarywriter = True
+    config.max_epochs = 1
+
+    curr_model_path = tmp_path / "foo.epoch=0-step=1.ckpt"
+    best_model_path = tmp_path / "foo.best.ckpt"
+    tb_path = tmp_path / "tensorboard"
+    csv_path = tmp_path / "csv_logs"
+
+    with ModelRunner(
+        config, output_dir=tmp_path, output_rootname="foo"
+    ) as runner:
+        runner.train([mgf_small], [mgf_small])
+
+    assert curr_model_path.is_file()
+    assert best_model_path.is_file()
+    assert tb_path.is_dir()
+    assert csv_path.is_dir()
+
+    curr_model_path.unlink()
+    best_model_path.unlink()
+    shutil.rmtree(tb_path)
+
+    with pytest.raises(FileExistsError):
+        with ModelRunner(
+            config, output_dir=tmp_path, output_rootname="foo"
+        ) as runner:
+            runner.train([mgf_small], [mgf_small])
+
+    assert not curr_model_path.is_file()
+    assert not best_model_path.is_file()
+    assert not tb_path.is_dir()
+    assert csv_path.is_dir()
+
+
+def test_log_metrics(monkeypatch, tiny_config):
```

Comment:
The git history here got messed up during a rebase, the only new addition here is now each log_metrics test case also tests aa_recall in addition to aa_precison.